src/components/Header.tsx

```diff
@@ -0,0 +1,66 @@
+import { FormEvent, useEffect, useRef, useState } from 'react';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todo: Omit<Todo, 'id'>;
+  onSubmit: (todo: Omit<Todo, 'id'>) => Promise<void>;
+  onChange: (value: string) => void;
+  onReset: () => void;
+  onError: (error: string) => void;
+};
+
+export const Header: React.FC<Props> = ({
+  todo,
+  onSubmit,
+  onChange,
+  onReset,
+  onError,
+}) => {
+  const titleField = useRef<HTMLInputElement>(null);
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    if (titleField.current) {
+      titleField.current.focus();
+    }
+  }, [onReset]);
```

Comment:
Why this dependency?
```suggestion
  }, []);
```